File: 01 - FUNDAMENTOS DE PROGRAMAÇÃO/C/Exercicio 4.c
/* 4 - Escreva um algoritmo para calcular as raízes de uma equação do segundo grau (Ax2 + Bx + C). Considere que o valor de A <> 0 e (B2– 4AC) > 0, = 0 e <>0. Escreva os valores na tela. */

/* Teste 2 raízes (a=1 b=-3 c=-10) //Teste 1 raíz (a=9 b=-12 c=4) //Teste 0 raíz (a=5 b=3 c=5) */

#include<stdio.h>
#include<math.h>

int main ()
{
    float a, b, c, delta, x1, x2;

    printf (" Digite o valor de A: ");
    scanf (" %f",&a);
    printf ("\n Digite o valor de B: ");
    scanf (" %f",&b);
    printf ("\n Digite o valor de C: ");
    scanf (" %f",&c);

    if (a == 0)
        printf ("\n O A deve ser diferente de 0! \n\n");
    else{

        delta = pow(b,2) -4*a*c;

        if (delta>0) {
            x1 = (-b -sqrt(delta))/(2*a);
            x2 = (-b +sqrt(delta))/(2*a);
            printf ("\n Equacao possui raizes reais: \n");
              printf ("\n x1 = %.2f\n\n",x1);
              printf (" x2 = %.2f\n\n",x2);
            }
            else{
                if (delta == 0){
                  x2= (-b +sqrt(delta))/(2*a);
                  printf ("\n Equacao possui uma raiz real \nx: %.2f\n\n",x2);
                }
                else{
                    printf ("\n Equacao nao possui raizes reais!!");
                }
            }
    }
    return 0;
}

File: 01 - FUNDAMENTOS DE PROGRAMAÇÃO/Dart/exercicio4.dart
// 4 - Escreva um algoritmo para calcular as raízes de uma equação do segundo grau (Ax2
// + Bx + C). Considere que o valor de A <> 0 e (B2– 4AC) > 0. Escreva os valores na tela.

/* Teste 2 raízes (a=1 b=-3 c=-10) //Teste 1 raíz (a=9 b=-12 c=4) //Teste 0 raíz (a=5 b=3 c=5) */

import 'dart:io';
import 'dart:math';

main(List<String> args) {
  print('Digite o valor de a: ');
  double a = double.parse(stdin.readLineSync()!);
  print('Digite o valor de b: ');
  double b = double.parse(stdin.readLineSync()!);
  print('Digite o valor de c: ');
  double c = double.parse(stdin.readLineSync()!);

  if (a == 0) {
    print('O valor de A deve ser diferente de 0!');
    return;
  }

  double delta = pow(b, 2) - 4 * a * c;
  if (delta > 0) {
    double x1 = (-b - sqrt(delta)) / (2 * a);
    double x2 = (-b + sqrt(delta)) / (2 * a);
    print('\nEquação possui raízes reais:');
    print('x1 = ${x1} \nx2 = ${x2}');
  } else if (delta == 0) {
    double x1 = (-b + sqrt(delta)) / (2 * a);
    print('\nEquação possui uma raiz real: \n${x1}');
  } else {
    print('\nEquação não possui raízes reais!');
  }
}
